src/ORM/Table.php

```diff
@@ -1680,9 +1680,7 @@ public function findOrCreate($search, callable $callback = null, $options = [])
             'defaults' => true
         ];
 
-        return $this->_executeTransaction(function () use ($search, $callback, $options) {
```

Comment:
I think the goal of this wrapping transaction was that the find and save were executed as one transaction. In repeatable-read or [higher isolation modes](https://en.wikipedia.org/wiki/Isolation_(database_systems)#Read_phenomena) this wrapping transaction aims to help prevent dirty and phantom reads.

With that said, I don't know how we solve the read phenomena and also fire the aftercommit event.